R/weight.R

```diff
@@ -41,33 +43,216 @@ weight <- function (x) {
+#' be a numeric Crunch variable, list of numeric Crunch variables or
+#' a character vector with the aliases of numeric Crunch variables.
+#'
+#' @return For the getter the variables which can be set as the dataset's weight. For
+#' the setter the dataset or variable catalog duly modified.
+#' @seealso [weight()] [makeWeight()]
+#' @name weightVariables
+#' @examples
+#' \dontrun{
+#' weightVariables(ds) <- ds$weight
+#' weightVariables(ds) <- list(ds$weight, ds$weight2)
+#' weightVariables(ds) <- NULL
+#' weightVariables(ds) <- c("weight", "weight2")
+#' #To append a weight to existing weights
+#' weightVariables(ds) <- c(weightVariables(ds), "weight3")
+#' #To remove a weight
+#' weightVariables(ds) <- setdiff(weightVariables(ds), "weight3")
+#' }
+#' @name weightVaraibles
+#' @aliases is.weightVariable<- weightVariables<-
+#' 
+#' @export
+NULL
+
+#' @rdname weightVariables
+#' @export
+setMethod("weightVariables", "CrunchDataset",
+    function(x) weightVariables(allVariables(x)))
+
+#' @rdname weightVariables
+#' @export
+setMethod("weightVariables", "VariableCatalog", function(x) {
     ## Get weight variable order
     ord <- VariableOrder(crGET(shojiURL(x, "orders", "weights")))
     ## Subset weight variable catalog with it
     vars <- x[ord]
     ## Return the names/aliases
     if (length(vars)) {
         return(sort(vapply(index(vars), vget(namekey(x)), character(1),
-            USE.NAMES=FALSE)))
+            USE.NAMES = FALSE)))
     } else {
         return(c())
     }
 })
 
+#' @rdname weightVariables
+#' @export
+setMethod("weightVariables<-", "CrunchDataset", function(x, value) {
+    weightVariables(allVariables(x)) <- value
+    x <- refresh(x)
+    return(x)
+})
+
+#' @rdname weightVariables
+#' @export
+setMethod("weightVariables<-", "VariableCatalog", function(x, value) {
+    modifyWeightVariables(x, value, type = "append")
+})
+
+#' @rdname modifyWeightVariables
+#' @export
+setMethod("modifyWeightVariables", "CrunchDataset", function(x, vars, type = "append") {
+    varcat <- allVariables(x)
+    modifyWeightVariables(varcat, vars, type)
+})
+
+#' @rdname modifyWeightVariables
+#' @export
+setMethod("modifyWeightVariables", "VariableCatalog", function(x, vars, type = "append") {
+    modifyWeightVariables(x, vars, type)
+})
+
+#' modifyWeightVariables
+#'
+#' Change which variables can be set as a dataset's weight.
+#'
+#' @param x a dataset or variable catalog
+#' @param vars Variables to add or remove  this can be a numeric Crunch variable,
+#' list of numeric Crunch variables or a character vector with the aliases of numeric Crunch variables.
+#' @param type a character string determining how the weightVariables
+#' will be modified:
+#' - `"append"` : add `vars` to the current weight variables
+#' - `"remove"` : remove `vars` from the current list of weight variables
+#' - `"replace"`: replace the current weight variables with `vars`
+#'
+#' @export
+modifyWeightVariables <- function(x, vars, type = "append") {
```

Comment:
This is... unnatural. You have `setMethod("modifyWeightVariables", ...)` but then you also have an exported function called `modifyWeightVariables`. I'd either delete the S4 method dispatch and do `if (is.dataset(x)) modifyWeightVariables(allVariables(x), ...)` pseudo-dispatching inside the function, or rename the function so that it doesn't clash with the methods.